cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/TimePartitionedFileSetDatasetAvroSink.java

```diff
@@ -86,9 +112,9 @@ public void prepareJob(BatchSinkContext context) {
     Map<String, String> runtimeArguments = context.getPluginProperties().getProperties();
     Map<String, String> sinkArgs = Maps.newHashMap();
     TimePartitionedFileSetArguments.setOutputPartitionTime(sinkArgs, context.getLogicalStartTime());
-    TimePartitionedFileSet sink = context.getDataset(runtimeArguments.get(TPFS_NAME), sinkArgs);
-    context.setOutput(runtimeArguments.get(TPFS_NAME), sink);
-    Schema avroSchema = new Schema.Parser().parse(runtimeArguments.get(SCHEMA));
+    TimePartitionedFileSet sink = context.getDataset(runtimeArguments.get(tpfsAvroSinkConfig.name), sinkArgs);
```

Comment:
Why are we using runtimeArguments? Shouldn't you directly use the config object?